lib/ansible/modules/cloud/vmware/vmware_dvs_portgroup_find.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r'''
+---
+module: vmware_dvs_portgroup_find
+short_description: Find portgroup(s) in a vmware environment
+description:
+- Find portgroup(s) based on different criteria such as dvswitch, vlan id or a string in the name.
+version_added: 2.9
+author:
+- David Martinez (@dx0xm)
+notes:
+- Tested on vSphere 6.5
+requirements:
+- python >= 2.7
+- PyVmomi
+options:
+  dvswitch:
+    description:
+    - Name of a dvswitch to look for.
+    type: str
+  vlanid:
+    description:
+    - vlan id can be any number between 1 and 4094.
+    - This search criteria will looks into vlan ranges to find possible matches.
+    required: false
+    type: int
+  namestr:
+    description:
+    - string to check inside the name of the portgroup.
+    type: str
+  show_uplink:
+    description:
+    - Show or hide uplink portgroups.
+    - Only relevant when C(vlanid) is supplied
```

Comment:
```suggestion
    - Only relevant when C(vlanid) is supplied.
```

Reply:
Thanks @Akasurde , will commit as soon as I clarify the previous one.